### PR TITLE
🪟 🎨  Add loading backdrop for sync catalog table

### LINKED ISSUE #14990: After editing a connection's replication settings and confirming a reset, the page doesn't refresh
Repro Steps:
1. Edit streams in a connection
2. Press "Save Changes"
3. Press "Save Connection" in the modal confirming the reset
4. See that the streams added/removed are still shown as pending.

Expected behavior:
- I would expect the contents of the page to refresh/reload and show the newly saved configuration settings.

### PR DESCRIPTION
**Pull request by dizel852**
## What
Closes: [#14990](https://github.com/airbytehq/airbyte/issues/14990)

## How
- Add loading backdrop for sync catalog table
- Refactor the `<Spinner />` component

CPT2208161812-1216x776

## Recommended reading order
1. `Spinner.*`
2. `LoadingBackdrop.*`

### CODE REVIEW

File: airbyte-webapp/src/components/LoadingBackdrop/LoadingBackdrop.module.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
+@use "../../scss/colors";
+
+.cover {
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/LoadingBackdrop/LoadingBackdrop.module.scss:3]: Stylelint will flag this, should be `%cover` and `@extend %cover`

File: airbyte-webapp/src/components/LoadingBackdrop/LoadingBackdrop.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+import Spinner from "../Spinner/Spinner";
+import styles from "./LoadingBackdrop.module.scss";
+
+interface LoadingBackdropProps {
+  isActive: boolean;
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/LoadingBackdrop/LoadingBackdrop.tsx:7]: I feel like `loading` may be a better term?

File: airbyte-webapp/src/components/Spinner/Spinner.tsx [hunk 3]
```diff
@@ -1,43 +1,14 @@
...
-  border-radius: 50%;
-  border: 4px solid ${({ theme }) => theme.primaryColor12};
-  position: relative;
-  animation: ${SpinAnimation} 1.5s linear 0s infinite;
-`;
-
-const BreakRec = styled.div<IProps>`
-  width: 13px;
-  height: 7px;
-  background: ${({ theme, backgroundColor }) => (backgroundColor ? backgroundColor : theme.whiteColor)};
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/Spinner/Spinner.tsx:31]: Did we need to preserve backgroundColor?

File: airbyte-webapp/src/components/LoadingBackdrop/LoadingBackdrop.module.scss (hunk 1 above)
Comment (dizel852) [reply airbyte-webapp/src/components/LoadingBackdrop/LoadingBackdrop.module.scss:3]: Oh, okay, will fix it

File: airbyte-webapp/src/components/Spinner/Spinner.tsx (hunk 3 above)
Comment (dizel852) [reply airbyte-webapp/src/components/Spinner/Spinner.tsx:31]: No actually - that was a hack, to make a "break" in the circle to match the background color.
image

File: airbyte-webapp/src/components/LoadingBackdrop/LoadingBackdrop.tsx (hunk 2 above)
Comment (dizel852) [reply airbyte-webapp/src/components/LoadingBackdrop/LoadingBackdrop.tsx:7]: Hm, yes, probably... Will fix it

**Review by krishnaglick — APPROVED** [review]

**Pull request merged**